server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNode.java

```diff
@@ -299,12 +304,8 @@ public Set<DiscoveryNodeRole> getRoles() {
         return roles;
     }
 
-    public boolean isCompatibleWithVersion(Version version) {
-        if (version.onOrAfter(RANGE_FLOAT_PROCESSORS_SUPPORT_VERSION)) {
-            return true;
-        }
-
-        return processorsRange == null && processors.isCompatibleWithVersion(version);
+    public boolean clusterHasRequiredFeatures(Predicate<NodeFeature> clusterHasFeature) {
+        return (processorsRange == null && processors.hasDecimals() == false) || clusterHasFeature.test(RANGE_FLOAT_PROCESSORS_SUPPORTED);
```

Comment:
Isn't this || inverted with respect to the original code?
(actually the original code has a duplicated test, but it short-circuit on version before testing processorRange - don't know if this was intended or not!)

Reply:
Here the predicate test is more expensive than a simple version comparison test, so I wanted to do that last.

Reply:
If it does not change the semantics (and I don't think it does) fine by me!